src/__tests__/mongo.test.ts

```diff
@@ -1,52 +1,67 @@
 import mongoose from "mongoose";
-import { connectToDatabaseViaEnvVar, ILog } from "../mongodb/mongo";
-
-import { Writable } from 'stream';
-import winston from 'winston';
+import {
+  afterEach,
+  beforeAll,
+  beforeEach,
+  describe,
+  expect,
+  it,
+  vi,
+} from "vitest";
+import {
+  connectToDatabaseViaEnvVar,
+  disconnectFromDatabase,
+  ILog,
+} from "../mongodb/mongo";
+
+import { Writable } from "stream";
+import winston from "winston";
 import { LambdaLogger } from "../logger";
 
 const fakeLogger = (): [ILog, string[]] => {
   let output: any[] = [];
-  const stream = new Writable()
+  const stream = new Writable();
   stream._write = (chunk, encoding, next) => {
-      output.push(chunk.toString());
-      next()
-  }
+    output.push(chunk.toString());
+    next();
+  };
 
-  const streamTransport = new winston.transports.Stream({ stream })
-  const logger = winston.createLogger({ transports: [streamTransport]})
+  const streamTransport = new winston.transports.Stream({ stream });
+  const logger = winston.createLogger({ transports: [streamTransport] });
   return [logger, output];
 };
 
-const mongooseIsConnected = () => {
-  return mongoose.connection.readyState === 1;
-};
-
-const ensureMongooseIsConnected = async () => {
-  if (!mongooseIsConnected()) {
-    await mongoose.connect(process.env.MONGODB_URI!);
-  }
-}
-
-const ensureMongooseIsDisconnected = async () => {
-  if (mongooseIsConnected()) {
-    await mongoose.disconnect();
-  }
-}
-
 describe("connectToDatabaseViaEnvVar", () => {
-  let originalMongoDBUri:(string|undefined)= "";
+  const mongooseIsConnected = () => {
+    return mongoose.connection.readyState === 1;
+  };
+
+  const ensureMongooseIsConnected = async () => {
+    if (!mongooseIsConnected()) {
+      console.log(process.env.MONGODB_URI);
+      await mongoose.connect(process.env.MONGODB_URI!);
+    }
+  };
+
+  const ensureMongooseIsDisconnected = async () => {
+    if (mongooseIsConnected()) {
+      await mongoose.disconnect();
+    }
+  };
+  let originalMongoDBUri: string | undefined = "";
 
   beforeAll(async () => {
     originalMongoDBUri = process.env.MONGODB_URI;
   });
 
   beforeEach(async () => {
+    vi.restoreAllMocks();
     process.env.MONGODB_URI = originalMongoDBUri;
     await ensureMongooseIsDisconnected();
   });
 
   afterEach(async () => {
+    vi.restoreAllMocks();
```

Comment:
Feels like we shouldn't need both of these.  Is there something we can add at the top level to make sure we're resetting to the same state before the start of each test? I don't think it makes a huge amount of difference whether it's setup or teardown that it happens in, but we should pick one.